### ISSUE TITLE
luac has problems with shellescape

### ISSUE THREAD

**Issue by ubunatic** [xolox/vim-lua-ftplugin#1]
I debugged the lua.vim script and found a problem on windows (7):
I reproduced it manually on the command line:

```
C:\somedir> luac -p 'C:/home/juve/workspace/test/main.lua'
```

produces an error:

```
luac: cannot open 'C:/home/juve/workspace/test/main.lua': Invalid argument
```

Using default double quotes works fine.

```
C:\somedir> luac -p "C:/home/juve/workspace/test/main.lua"
```

The problem derives from vim's shellescape (I guess):

```
function xolox#lua#checksyntax
line 18: execute 'silent make!' compiler_args shellescape(expand('%'))
```

The status line in Vim shows:

```
lua.vim 0.6.27: Syntax error on line 0: luac: cannot open 'C:/home/juve/workspace/test/main.lua': Invalid argument
```

**Comment by ubunatic** [comment]
OK, found the problem. I did set `shellslash`.
`:set noshellslash` in `.vimrc` solved the problem.

**Comment by xolox** [comment]
Hi Uwe,

I've had so many bug reports about the [shellslash](http://vimdoc.sourceforge.net/htmldoc/options.html#%27shellslash%27) option breaking the [shellescape()](http://vimdoc.sourceforge.net/htmldoc/eval.html#shellescape%28%29) function that I now consider it a bug when my Vim plug-ins don't work around this issue :-). Anyway, I've just released version 0.6.28 of the Lua file type plug-in which should work regardless of the shellslash setting. Thanks for the feedback!

 - Peter
